### PR TITLE
Deploy infrastructure resources to AWS region us-west-2 and update to current project architecture

### PR DESCRIPTION
**Pull request by rocketnova**
## Ticket

- https://wicmtdp.atlassian.net/browse/PRP-193
- https://wicmtdp.atlassian.net/browse/PRP-161
- https://wicmtdp.atlassian.net/browse/PRP-194

- Change regions from `us-west-1` to `us-west-2`
- Modify the env-template to create 1 database, 1 service cluster, and 3 services
- Update some CI/CD from the [infra template](https://github.com/navapbc/template-infra/tree/3d7724a179da0d13b8a5d981890f428bec6f6005) (cd.yml, Makefile, bin)
- Move IAM module to be called by the `infra/accounts/account/` module
- Break out database resources into separate `infra/modules/database` module
- Break out random password generation into separate `infra/modules/random_password` module
- Break out ECS cluster into separate `infra/modules/service_cluster` module
- Update IAM permissions for `wic-prp-eng` user group
- Rename some variables that were still called `api` 

This PR started out as just migrating our resources from `us-west-1` to `us-west-2`, but as we deleted all the resources and re-created them, it didn't make sense to re-create them in the wrong pattern when we knew we were about to need them to be different.

So, in addition to the region change, we also modified the env-template to support our current project architecture (1 database, 1 service cluster, 3 services). To do this, we broke out some of the resources that were previously grouped into the service module into their own separate modules. We renamed our resources to ensure that each environment had its own dedicated resources.

These changes have all been deployed to our AWS account. To confirm that all of the resources exist and are configured properly. Please check all of the following against the terraform files.

### build-repository
- ecr: https://us-west-2.console.aws.amazon.com/ecr/repositories?region=us-west-2

### cluster: 
- ecs cluster: https://us-west-2.console.aws.amazon.com/ecs/v2/clusters?region=us-west-2

### database
- rds cluster: https://us-west-2.console.aws.amazon.com/rds/home?region=us-west-2#databases:
- rds instance: same as above
- ssm password: https://us-west-2.console.aws.amazon.com/systems-manager/parameters/?region=us-west-2&tab=Table#
- backup plan: https://us-west-2.console.aws.amazon.com/backup/home?region=us-west-2#/backupplan
- backup vault: https://us-west-2.console.aws.amazon.com/backup/home?region=us-west-2#/backupvaults
- iam role "postgresql-backup": https://us-east-1.console.aws.amazon.com/iamv2/home?region=us-west-2#/roles
- iam role "rds-enhanced monitoring": same as above
- iam policy "db-access": https://us-east-1.console.aws.amazon.com/iamv2/home?region=us-west-2#/policies

### service
- ecs service: service for each cluster
- ecs task definition: https://us-west-2.console.aws.amazon.com/ecs/v2/task-definitions?region=us-west-2
- ecs alb: https://us-west-2.console.aws.amazon.com/ec2/home?region=us-west-2#LoadBalancers: 
- ecs alb listener: listener tab for each alb
- ecs alb target group: https://us-west-2.console.aws.amazon.com/ec2/home?region=us-west-2#TargetGroups:
- cloudwatch log group: https://us-west-2.console.aws.amazon.com/cloudwatch/home?region=us-west-2#logsV2:log-groups
- iam role "task-executor": https://us-east-1.console.aws.amazon.com/iamv2/home?region=us-west-2#/roles
- security group alb: https://us-west-2.console.aws.amazon.com/ec2/home?region=us-west-2#SecurityGroups:
- security group app: same as above

### CODE REVIEW

File: infra/app/envs/dev/main.tf [hunk 1]
```diff
@@ -8,14 +8,14 @@ locals {
...
-  region = "us-west-1"
+  region = "us-west-2"
   # Add environment specific tags
   tags = merge(module.project_config.default_tags, {
     environment = local.environment_name
     description = "Application resources created in dev environment"
   })
 
-  tfstate_bucket = "wic-prp-636249768232-us-west-1-tf-state"
+  tfstate_bucket = "wic-prp-636249768232-us-west-2-tf-state"
```
Comment (microwavenby) [inline infra/app/envs/dev/main.tf:18]: I'm sure these buckets already exist, but if this were a project with more s3 buckets, i'd prefer to have the informational parts of the bucket name (`tf-state`) in the front and the "uniqueness" parts of the bucket name at the end

File: infra/modules/database/main.tf [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+    value = "1"
+  }
+}
+
+################################################################################
+# IAM role for user access
+################################################################################
+resource "aws_iam_policy" "db_access" {
+  name        = "${var.database_name}-db-access"
+  description = "Allows access to the database instance"
```
Comment (microwavenby) [inline infra/modules/database/main.tf]: This feels like an inaccurate description

IAM isn't required to access the database.
This is a permission to _modify_ and/or administer the database instance...

So, maybe
```terraform
   description = "Allows administration of the database instance"
```

File: infra/modules/iam/main.tf [hunk 3]
```diff
@@ -20,112 +20,236 @@ resource "aws_iam_group_policy" "wic-prp-eng" {
...
+    ]
   }
   statement {
-    sid    = "KMSServices"
+    sid    = "KMS"
     effect = "Allow"
     actions = [
+      "kms:DescribeKey",
       "kms:CreateAlias",
       "kms:Decrypt",
```
Comment (microwavenby) [inline infra/modules/iam/main.tf:171]: This list of actions doesn't allow encryption.. 
Are y'all sure this set of perms will allow someone to put new files into the S3 bucket, with encryption using that KMS key?

Comment (rocketnova) [reply infra/modules/iam/main.tf:171]: Good question! I'm not sure. I know these permissions allow a user (one of us) in the user group to create and destroy environments, including the S3 buckets. 

Do we need to put files into the S3 bucket? Is this in reference to how we're going to handle doc upload?

**Comment by microwavenby** [comment]
Sure is! If it doesn't apply, ignore it and I'll approve 🙃

**Comment by rocketnova** [comment]
> Sure is! If it doesn't apply, ignore it and I'll approve 🙃

@microwavenby Yah, this PR doesn't cover anything to do with doc upload, so it's a good concern for @aplybeah's upcoming PR on doc upload

**Review by microwavenby — APPROVED** [review]

**Review by aplybeah — APPROVED** [review]

**Pull request merged**